### ISSUE TITLE
Typescript error when using Nitro plugin in nuxt 3 project due to missing type for "render:html" in `NitroRuntimeHooks` interface

### ISSUE THREAD

**Issue by kimbaudi** [unjs/nitro#1337]
### Environment

- Operating System: Windows_NT
- Node Version:     v18.12.0
- Nuxt Version:     3.5.3
- Nitro Version:    2.5.1
- Package Manager:  yarn@1.22.19
- Builder:          vite
- User Config:      devtools, typescript
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

I created a minimal reproduction at https://github.com/kimbaudi/nuxt-nitro.

To reproduce the issue:

```shell
git clone https://github.com/kimbaudi/nuxt-nitro

npm install
// or yarn install

npx tsc
// or yarn tsc
```

### Describe the bug

Following the nuxt doc for extending HTML template (https://nuxt.com/docs/getting-started/views#advanced-extending-the-html-template), I created `server/plugins/extend_html.ts` with the following content:

```ts
export default defineNitroPlugin((nitroApp) => {
  nitroApp.hooks.hook("render:html", (html, { event }) => {
    html.head.push(`<meta name="author" content="Paul Kim" />`);
    console.log(event.node.res.statusCode);
  });
});
```

And I get the following typescript errors:

- Argument of type '"render:html"' is not assignable to parameter of type 'HookKeys<NitroRuntimeHooks>'.ts(2345)
- Property 'head' does not exist on type 'Partial<RenderResponse>'.ts(2339)

Looking at `node_modules/nitropack/dist/runtime/types.d.ts`, it seems that `NitroRuntimeHooks` interface is missing `"render:html"` property

![image](https://github.com/nuxt/nuxt/assets/1864092/e9b82540-b89f-4dd2-887f-979e517e5091)

In order to resolve this issue, the `NitroRuntimeHooks` interface should be updated to include "render:html" property. (https://github.com/unjs/nitro/blob/main/src/runtime/types.ts#L23)

### Logs

```sh
-script
$ npx tsc
server/plugins/extend-html.ts:2:23 - error TS2345: Argument of type '"render:html"' is not assignable to parameter of type 'HookKeys<NitroRuntimeHooks>'.

2   nitroApp.hooks.hook("render:html", (html, { event }) => {
                        ~~~~~~~~~~~~~

server/plugins/extend-html.ts:3:10 - error TS2339: Property 'head' does not exist on type 'Partial<RenderResponse>'.

3     html.head.push(`<meta name="author" content="Paul Kim" />`);
           ~~~~


Found 2 errors in the same file, starting at: server/plugins/extend-html.ts:2
```

**Comment by kimbaudi** [comment]
Since this bug report is related to Nuxt framework, I opened an issue at https://github.com/nuxt/nuxt/issues/21708

**Comment by danielroe** [comment]
This should be resolved in next Nuxt release :+1:

**Comment by kimbaudi** [comment]
I can confirm that this issue is resolved in nuxt v3.6.1
